spring-boot/pom.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>infinispan-simple-tutorials-spring-boot</artifactId>
+    <parent>
+        <relativePath>../pom.xml</relativePath>
+        <version>1.0.0-SNAPSHOT</version>
+        <groupId>org.infinispan.tutorial.simple</groupId>
+        <artifactId>infinispan-simple-tutorials</artifactId>
+    </parent>
+
+    <name>Infinispan Simple Tutorials: Spring Boot</name>
+
+    <properties>
+        <spring-boot.version>1.4.0.RELEASE</spring-boot.version>
+        <infinispan.starters.version>1.0.0-SNAPSHOT</infinispan.starters.version>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-starter-parent</artifactId>
+                <version>${spring-boot.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.infinispan</groupId>
+            <artifactId>inifinispan-spring-boot-starter</artifactId>
+            <version>${infinispan.starters.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.infinispan</groupId>
+                    <artifactId>infinispan-core</artifactId>
+                </exclusion>
```

Comment:
It's not about self-contained, it's about what you have to do if you want "more". I think you're asking the wrong question: Spring Boot starter have no "preference". The only reason why it exist is to serve the (infinispan) user and not go in their way. 

If the vast majority of your users use the embedded mode then by all means provide that by default (but you shouldn't have to exclude it to use the client mode). Have you looked at our JMS support? It's the exact same thing and has a lot maturity so maybe [you should look there for inspiration](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jms/artemis/ArtemisConnectionFactoryFactory.java#L95-L105)). So, we look in the classpath: if we have both, we prefer the embedded mode but there is a property the user can set to specify in which mode they want to run (for JMS that's `spring.artemis.embedded.enabled`).

Long story short: it does not matter if the starter is self contained or not. What matters is what the starter should logically do by default. We usually go with client/server because adding the "whole server" by default can be quite involved (but I reckon that caching libraries have a lower footprint). If the vast majority of your users use infinispan in X mode, that's what the starter should do by default.